assets/scripts/dialogs/GeotagDialog.jsx

```diff
@@ -285,104 +277,69 @@ class GeotagDialog extends React.Component {
+            zoom={zoom}
+            onClick={geocodeAvailable ? handleMapClick : null}
+            useFlyTo={true}
+            id="map"
+          >
+            <TileLayer attribution={MAP_ATTRIBUTION} url={tileUrl} />
+            <ZoomControl
+              zoomInTitle={intl.formatMessage({
+                id: 'dialogs.geotag.zoom-in',
+                defaultMessage: 'Zoom in'
+              })}
+              zoomOutTitle={intl.formatMessage({
+                id: 'dialogs.geotag.zoom-out',
+                defaultMessage: 'Zoom out'
+              })}
+            />
+
+            {renderPopup && (
+              <LocationPopup
+                position={markerLocation}
+                label={label}
+                isEditable={geocodeAvailable && canEditLocation()}
+                isClearable={geocodeAvailable && canClearLocation()}
+                handleConfirm={(e) => {
+                  handleConfirmLocation(e)
+                  closeDialog()
+                }}
+                handleClear={(e) => {
+                  handleClearLocation(e)
+                  closeDialog()
+                }}
+              />
+            )}
 
-              {this.state.renderPopup && (
-                <LocationPopup
-                  position={this.state.markerLocation}
-                  label={this.state.label}
-                  isEditable={
-                    this.state.geocodeAvailable && this.canEditLocation()
-                  }
-                  isClearable={
-                    this.state.geocodeAvailable && this.canClearLocation()
-                  }
-                  handleConfirm={(e) => {
-                    this.handleConfirmLocation(e)
-                    closeDialog()
-                  }}
-                  handleClear={(e) => {
-                    this.handleClearLocation(e)
-                    closeDialog()
-                  }}
-                />
-              )}
-
-              {this.state.markerLocation && (
-                <Marker
-                  position={this.state.markerLocation}
-                  onDragEnd={this.handleMarkerDragEnd}
-                  onDragStart={this.handleMarkerDragStart}
-                  draggable={this.state.geocodeAvailable}
-                />
-              )}
-            </Map>
-          </div>
-        )}
-      </Dialog>
-    )
-  }
-}
-
-function mapStateToProps (state) {
-  return {
-    street: state.street,
-    markerLocation: state.map.markerLocation,
-    addressInformation: state.map.addressInformation,
-    userLocation: state.user.geolocation.data,
-    dpi: state.system.devicePixelRatio
-  }
-}
-
-const mapDispatchToProps = {
-  setMapState,
-  addLocation,
-  clearLocation,
-  saveStreetName
+            {markerLocation && (
+              <Marker
+                position={markerLocation}
+                onDragEnd={handleMarkerDragEnd}
+                onDragStart={handleMarkerDragStart}
+                draggable={geocodeAvailable}
```

Comment:
This is a good question to raise, in part because the second point is a real issue. If Pelias does not have data for a point, it tries to find the nearest result, and it may not be what the user wanted.

The additional work on top of this is that all of our display values assume the presence of a human-readable label. We would need to go through and find out where to display lat,lng values instead, if the label isn't present. And if a person hits a point close to a geocodable point, and that _is_ what they mean, how close do they need to be before we assume they want a geocadable point or a lat,lng point? Is a lat,lng point always available or do we make that decision for them?

(re: your first point, Leaflet definitely can return a lat,lng without a geocoder available. It has to know because map tiles can be available without a geocoding service.)

File an issue for this if you're interested in taking this on.